### PR TITLE
feat(frontend): add referral modal

### PR DESCRIPTION
**Pull request by boriskubrik**
- Adds a referral modal. 
- Places links in the Menu and in the footer.
- The URL does not contain `feeBps`, sticking to default. UI shows 1% because 50% goes to the protocol.
- Uses context to keep track of the state as we have the modal in multiple places.
- Triggers the share action on mobile, copies to clipboard on desktop.
- Shows toast after successful copy on desktop.
- Shows connect wallet if the user is not connected.

### Testing
- [x] Sharing works on mobile using MetaMask
- [x] Copy works on desktop
- [x] Referral address and fee set in localstorage after using the link
- [x] Connect wallet state works

### Screenshot 
image

### CODE REVIEW

File: packages/frontend/src/modals/ReferralModal/ReferralModal.tsx [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+import { useAccount } from 'wagmi';
+
+const ReferralModal: FunctionComponent = () => {
+  const { isReferralModalOpen, closeReferralModal } = useReferralModal();
+  const { address } = useAccount();
+
+  const generateReferralLink = () => {
+    if (!address) return null;
+
+    const feeBps = 20;
```
Comment (tommy0x) [inline packages/frontend/src/modals/ReferralModal/ReferralModal.tsx:15]: This should be 200 if I'm correct. 1 BPS is 0.01%

**Review by tommy0x — CHANGES_REQUESTED** [review]

**Review by immutablegoblin — COMMENTED** [review]
Looks good, just needs bps adjustment as noted by Tommy

**Comment by boriskubrik** [comment]
Need to hold on until the URL structure is finalized. #479

**Comment by xykota** [comment]
Let's make this ref link not include the fee bps. Then we'll have the default fee when there is a partner at 40 bps and without a partner 20 bps.

Then in the next iteration we will let the user customize the fee bps and reflect in the text they are sharing it down the middle with the protocol.

ref #479

**Comment by boriskubrik** [comment]
Rebased, updated the URL, tested and it works. Ready for review.

image

**Comment by immutablegoblin** [comment]
It would be nice not to have the linebreak when the modal is at full width if possible?

Working well though.

image

**Comment by boriskubrik** [comment]
@immutablegoblin switched to `break-all`.

image

**Review by immutablegoblin — APPROVED** [review]

**Pull request merged**